### ISSUE TITLE
multiple 'contains' on the same attribute (AND)

### ISSUE THREAD

**Issue by tibotiber** [balderdashy/waterline#648]
Hi! 
What's the status on overloaded `contains` modifier? (#176)
I've been trying to use several `contains` modifiers on the same attribute but nothing seems to work. Below is what I've tried so far...
```js
FishVariant.find({skuPicker: {contains: ['a', 'b']}})...
FishVariant.find({skuPicker: [{contains: 'a'}, {contains: 'b'}]})...
FishVariant.find([{skuPicker: {contains: 'a'}}, {skuPicker: {contains: 'b'}}])...
FishVariant.find().where({skuPicker: {contains: 'a'}}).where({skuPicker: {contains: 'b'}})...
```
What I am trying to get is a record for which the skuPicker attribute contains both 'a' and 'b' (just to be clear). 

You have a similar post on your [google group](https://groups.google.com/forum/#!searchin/sailsjs/contains/sailsjs/5q7TRyoXIhw/wUPTsG0PoP0J).

**Comment by dmarcelino** [comment]
Hi @tibotiber, I don't think there is support for this yet. It would be good to to be able to do something like: 
``` javascript
FishVariant..find({
          and: [
            { skuPicker: { contains: 'a' } },
            { skuPicker: { contains: 'b' } }
          ]
        })
```
I'm labelling it as feature request.

@particlebanana, @devinivy, @tjwebb, @randallmeeker, guys what do you think of adding this to v0.11? It seems like something it should be supported.

**Comment by randallmeeker** [comment]
Is this only with ```AND```, because I use it currently with ```OR``` and it works.

**Comment by dmarcelino** [comment]
Just `AND`. My example is actually based on the `OR` example.

**Comment by randallmeeker** [comment]
Then yes, it should be on .11 !

**Comment by tibotiber** [comment]
Great! Thanks guys! :)

**Comment by dmarcelino** [comment]
We can probably take a look at #666 (`NOT LIKE` /  `NOT CONTAINS`) while we are adding this, I believe it's around the same area.

**Comment by sailsbot** [comment]
Thanks for posting, @tibotiber.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by dmarcelino** [comment]
@particlebanana, @devinivy, we should reopen this one as having an `AND` greatly improves the querying capabilities of waterline.

**Comment by tibotiber** [comment]
Thanks @dmarcelino and others!

**Comment by sailsbot** [comment]
Thanks for posting, @tibotiber.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by cmddavid** [comment]
@tibotiber, how did you end up working around this problem? I'm looking for a temporary workaround until 0.11.

**Comment by tibotiber** [comment]
@cmddavid, it's a pretty ugly and manual quick fix since I'm waiting for 0.11. I've worked around that by finding only the first part of the `AND` and then filtering the result set to check other parts (more than 2). My code is more complex than that, but I've put together (a bit too fast maybe) a simplified version below. 

```js
FishVariant.find({skuPicker: {contains: options.search.split('+').slice(0, 1)}}).exec(function(err, found) {
  // [...]
  var containsAllCriteria = function(item) {
    var noMatch = _.reject(options.search.split('+').slice(1), function(criteria) {
      return item.skuPicker.toLowerCase().indexOf(criteria.toLowerCase()) > -1;
    });
    return noMatch.length === 0;
  };
  found = _.filter(found, containsAllCriteria);
  // [...]
});
```

Hope this can help you.

**Comment by cmddavid** [comment]
@tibotiber Thank you for your help, this is a decent solution for now. Only thing i'm worried about now is how to fix the pagination which the filter will break. Because the filter doesn't give a consistent amount of results per step. But since i'm using infinite scrolling this should not be a big problem.

**Comment by martskins** [comment]
Is this still a thing in waterline 0.13.x? I have a pretty simple implementation of this in a fork of `waterline-sequel`. It will only work when passing the and in a single `where`, chaining them won't work. But considering waterline 0.13.x does not support chained `where`s it might be good enough. I still have to solve this for `sails-mongo` though.
